forums/forums-library/src/main/java/org/silverpeas/components/forums/service/ForumsDAO.java

```diff
@@ -675,14 +677,11 @@ public static void deleteForum(Connection con, ForumPK forumPK) throws SQLExcept
 
 
   public static void deleteAllForums(Connection con, String instanceId) throws SQLException {
-    List<Integer> forumIds = new ArrayList<>(getAllForumsByInstanceId(con, instanceId));
-    StringBuilder listOfIds = new StringBuilder("(");
-    for (int i = 0; i < forumIds.size() - 1; i++) {
-      listOfIds.append("'").append(forumIds.get(i)).append("', ");
-    }
-    listOfIds.append("'").append(forumIds.get(forumIds.size() - 1)).append("')");
-    if (listOfIds.length() > 0) {
-      try (PreparedStatement statement = con.prepareStatement(FORUM_RIGHTS_DELETION + listOfIds.toString())) {
+    Collection<Integer> forumIds = getAllForumsByInstanceId(con, instanceId);
+    if (!forumIds.isEmpty()) {
+      String listOfIds =
+          forumIds.stream().map((i) -> "'" + i + "'").collect(Collectors.joining(",", "(", ")"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "i" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1611)